### PR TITLE
Subjob inputdata references master job

### PR DESCRIPTION
**Pull request by mesmith75**
This only works with `SplitByFiles`. When subjobs are created their `inputdata` now points to the `inputdata` of the master job. Should save a bit of space.

### CODE REVIEW

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from GangaDirac.Lib.Backends.DiracUtils import get_result
 from GangaCore.GPIDev.Lib.GangaList.GangaList import GangaList, makeGangaListByRef
 from GangaCore.GPIDev.Adapters.IGangaFile import IGangaFile
+import GangaCore.GPI as GPI
```
Comment (egede) [inline ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Looks odd. Why do we want to use the GPI? Isn't that only for the end user?

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 2]
```diff
@@ -55,7 +57,7 @@ class LHCbDataset(GangaDataset):
                       'getLFNs', 'getFileNames', 'getFullFileNames',
                       'difference', 'isSubset', 'isSuperset', 'intersection',
                       'symmetricDifference', 'union', 'bkMetadata',
-                      'isEmpty', 'hasPFNs', 'getPFNs']  # ,'pop']
+                      'isEmpty', 'hasPFNs', 'getPFNs', 'setReference', 'createIndex']  # ,'pop']
```
Comment (egede) [inline ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Should these new methods be exposed to the end user?

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 3]
```diff
@@ -110,20 +112,37 @@ def isFileTest(_file):
...
+
     def __getitem__(self, i):
         '''Proivdes scripting (e.g. ds[2] returns the 3rd file) '''
         #this_file = self.files[i]
         # print type(this_file)
         # return this_file
         # return this_file
         # return this_file
+        if self.ref:
+            return GPI.jobs(self.ref).inputdata[self.files][i]
```
Comment (egede) [inline ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: So is this really the correct way to do this?

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 4]
```diff
@@ -110,20 +112,37 @@ def isFileTest(_file):
...
         logger.debug("Dataset Created")
 
+    def setReference(self, jobNo, indices):
+        '''A function to set up the references of a subjob's dataset to the master job'''
+        self.ref = jobNo
+        self.files = indices
+
+    def createIndex(self):
+        '''Populate a list of the dataset's LFNs for some quick sorting'''
+        self.lfnList = [df.lfn for df in self.files]
```
Comment (egede) [inline ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py:179]: Will this create a very large list in memory that we then never garbage collect? Is it a problem?

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 1 above)
Comment (mesmith75) [reply ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: I agree it is not ideal. But `LHCbDataset` should exist as a standalone object so this was the only way I could think of to access a job on the registry.

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 2 above)
Comment (mesmith75) [reply ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Ah no, that was for my testing. I'll remove that.

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 3 above)
Comment (mesmith75) [reply ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: I couldn't think of a better way...but there probably is one.

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 4 above)
Comment (mesmith75) [reply ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py:179]: Yes. I should get the splitter to clear this after splitting.

File: ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py (hunk 1 above)
Comment (mesmith75) [reply ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Found a better way.

**Review by egede — APPROVED** [review]

**Pull request merged**